From: JungRae Kim <jungrae_kim@tmax.co.kr>
Date: Mon, 29 Mar 2021 13:57:58 +0900
Subject: [feat] Add tos session

 - Add files needed for tos session
 - Comment out files for gnome & gnome-xorg session

[IMS] N/A
---
 data/meson.build               | 19 ++++++++++++++++---
 data/tos.desktop.in.in         | 11 +++++++++++
 data/tos.json.in               | 10 ++++++++++
 data/tos.session.desktop.in.in |  5 +++++
 meson_post_install.py          |  6 +++---
 5 files changed, 45 insertions(+), 6 deletions(-)
 create mode 100644 data/tos.desktop.in.in
 create mode 100644 data/tos.json.in
 create mode 100644 data/tos.session.desktop.in.in

diff --git a/data/meson.build b/data/meson.build
index e02b80d..b3a02a9 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,8 +1,9 @@
 desktop_plain = 'gnome'
 
 desktops = [
-  desktop_plain,
-  'gnome-xorg'
+  #  desktop_plain,
+  #  'gnome-xorg',
+  'tos'
 ]
 
 shell_component = {
@@ -70,7 +71,8 @@ endforeach
 
 sessions = [
   'gnome',
-  'gnome-dummy'
+  #  'gnome-dummy',
+  'tos'
 ]
 
 foreach session: sessions
@@ -191,3 +193,14 @@ install_data(
   data,
   install_dir: session_pkgdatadir
 )
+
+tosmode_conf = configuration_data()
+tosmode_conf.set('uuid_suffix', 'tmax-shell-extensions')
+tosmode = 'tos.json'
+configure_file(
+  input: tosmode + '.in',
+  output: tosmode,
+  install: true,
+  install_dir: join_paths(session_datadir, 'gnome-shell', 'modes'),
+  configuration: tosmode_conf
+)
diff --git a/data/tos.desktop.in.in b/data/tos.desktop.in.in
new file mode 100644
index 0000000..803f85e
--- /dev/null
+++ b/data/tos.desktop.in.in
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Name[en_GB]=Tmax OS
+Name[ko]=토스
+Name=Tmax OS
+Comment[en_GB]=This session logs you into Tmax OS
+Comment[ko]=이 세션을 사용하면 토스에 로그인합니다
+Comment=This session logs you into Tmax OS
+Exec=env GNOME_SHELL_SESSION_MODE=tos gnome-session "$@"
+TryExec=gnome-session
+Type=Application
+DesktopNames=TOS;GNOME;
diff --git a/data/tos.json.in b/data/tos.json.in
new file mode 100644
index 0000000..566bc93
--- /dev/null
+++ b/data/tos.json.in
@@ -0,0 +1,10 @@
+{
+    "parentMode": "user",
+    "enabledExtensions":[ "alternate-tab@@uuid_suffix@",
+                          "desktop-icons-ng@@uuid_suffix@",
+                          "dash-to-panel@jderose9.github.com",
+                          "notification-center@@uuid_suffix@",
+                          "arc-menu@tmax",
+                          "appindicator@@uuid_suffix@"
+                        ]
+}
diff --git a/data/tos.session.desktop.in.in b/data/tos.session.desktop.in.in
new file mode 100644
index 0000000..5eca64b
--- /dev/null
+++ b/data/tos.session.desktop.in.in
@@ -0,0 +1,5 @@
+[GNOME Session]
+Name[en_GB]=Tmax OS
+Name[ko]=토스
+Name=Tmax OS
+RequiredComponents=org.gnome.Shell;org.gnome.SettingsDaemon.A11ySettings;org.gnome.SettingsDaemon.Clipboard;org.gnome.SettingsDaemon.Color;org.gnome.SettingsDaemon.Datetime;org.gnome.SettingsDaemon.Housekeeping;org.gnome.SettingsDaemon.Keyboard;org.gnome.SettingsDaemon.MediaKeys;org.gnome.SettingsDaemon.Mouse;org.gnome.SettingsDaemon.Power;org.gnome.SettingsDaemon.PrintNotifications;org.gnome.SettingsDaemon.Rfkill;org.gnome.SettingsDaemon.ScreensaverProxy;org.gnome.SettingsDaemon.Sharing;org.gnome.SettingsDaemon.Smartcard;org.gnome.SettingsDaemon.Sound;org.gnome.SettingsDaemon.Wacom;org.gnome.SettingsDaemon.XSettings;
diff --git a/meson_post_install.py b/meson_post_install.py
index e2e352c..c700c6c 100644
--- a/meson_post_install.py
+++ b/meson_post_install.py
@@ -21,6 +21,6 @@ dst_dir = os.path.join(install_root, 'wayland-sessions')
 if not os.path.exists(dst_dir):
   os.makedirs(dst_dir)
 
-src = os.path.join(install_root, 'xsessions', 'gnome.desktop')
-dst = os.path.join(dst_dir, 'gnome.desktop')
-shutil.copyfile(src, dst)
+#src = os.path.join(install_root, 'xsessions', 'gnome.desktop')
+#dst = os.path.join(dst_dir, 'gnome.desktop')
+#shutil.copyfile(src, dst)
